### PR TITLE
Handle timeouts in monitor queries in the UI

### PR DESCRIPTION
**Pull request by jpople**
Closes HJ-188

### Description Of Changes

Adds handling and feedback for when async monitor queries take longer than 5s to resolve:
* when loading projects for the project selection screen times out, monitor will be saved with `databases: []` (scoped to all projects)
* when saving a monitor times out, user receives a notice that the monitor will continue to be saved in the background

Screen recs (note that these have the timeout delay adjusted as described in the testing steps)

https://github.com/user-attachments/assets/3b3ad0ed-3240-4247-bece-e6eaccc9b078

https://github.com/user-attachments/assets/fbbd242d-cf3e-4664-a38a-2609d8bea3b8

### Steps to Confirm

This is tricky to test-- I've had best results by setting my browser throttling to "GPRS" and temporarily adjusting the timeout delays to be shorter (e.g. 500 ms).

#### Saving a monitor

1. Create a monitor as normal on a regular-speed connection
2. Behavior should be unchanged (monitor should save, should see success toast then modal closes)
3. Adjust `TIMEOUT_DELAY` of 5000 ms in `ConfigureMonitorModal.tsx` to something shorter and slow down your connection
5. Create a monitor
6. Save the monitor
7. When the timeout delay is up and the save query is still loading, should close out of modal and info toast should appear clarifying that the monitor is being saved in the background.

#### Project list

1. Create a monitor as normal on a regular-speed connection
2. Wait 5 seconds on project selection screen (projects should finish loading)
3. Behavior should be unchanged, no toast and modal should remain open
4. Adjust `TIMEOUT_DELAY` in `useCumulativeGetDatabases.ts` and slow down your connection
5. Refresh the page

RTK will cache the results of the project list query when it's called on the main "Monitors" tab and then it'll load instantly inside the modal, so we want to temporarily disable this to make it take longer to load:

6. In `ConfigureMonitorModal.tsx` comment out lines 44-50: 
```tsx
  // const { data: databases } = useGetDatabasesByConnectionQuery({
  //   page: 1,
  //   size: 25,
  //   connection_config_key: integration.key,
  // });

  // const databasesAvailable = !!databases && !!databases.total;
```
7. Then set the `databasesAvailable` prop to `true` in the ConfigureMonitorDatabasesForm (line 102):
```tsx
      {formStep === 0 && (
        <ConfigureMonitorForm
          monitor={monitor}
          onClose={onClose}
          onAdvance={onAdvance}
          onSubmit={handleSubmit}
          isSubmitting={isSubmitting}
          // databasesAvailable={databasesAvailable}
          databasesAvailable
          integrationOption={integrationOption}
        />
      )}
```
8. Click into "Data discovery" tab
9. Create a new monitor
10. When project list fails to load before the delay, monitor should be saved with `databases: []` and a toast should appear explaining what happened

### Pre-Merge Checklist

* [x] Issue requirements met
* [x] All CI pipelines succeeded
* [x] `CHANGELOG.md` updated
* Followup issues:
  * [ ] Followup issues created (include link)
  * [x] No followup issues
* Database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!
  * [x] No migrations
* Documentation:
  * [ ] Documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] Documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] If there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
  * [x] No documentation updates required

### CODE REVIEW

File: clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorDatabasesForm.tsx [hunk 1]
```diff
@@ -58,6 +76,10 @@ const ConfigureMonitorDatabasesForm = ({
 
   const saveIsDisabled = !allSelected && selected.length === 0;
 
+  if (initialIsLoading) {
+    return <FidesSpinner my={12} />;
```
Comment (gilluminate) [inline clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorDatabasesForm.tsx:80]: is this meant to center it? If so, wouldn't using auto margins be a safer bet there?
```suggestion
    return <FidesSpinner my="auto" />;
```

**Review by gilluminate — APPROVED** [review]
Approved with one minor suggestion.

File: clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorDatabasesForm.tsx (hunk 1 above)
Comment (jpople) [reply clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorDatabasesForm.tsx:80]: This is just to give it some room to breathe when it's the only thing in the modal.

No margin or auto:
Screenshot 2024-11-20 at 11 29 15

Constant margin:
Screenshot 2024-11-20 at 11 29 30

Comment (gilluminate) [reply clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorDatabasesForm.tsx:80]: Oh man. My bad. I confused my `x` and `y` lol.

**Pull request merged**